### DISCUSSION TITLE
Question about Trimanic StepSticks

### DISCUSSION THREAD

**Discussion by phil-barrett** [terjeio/grblHAL#235] [General]
[edit: I realized I misspelled Trinamic. No way to change. Sigh]

I have been looking at the TMC based products and have seen some discussion of 3.3V use of Trinamic stepsticks.  There is a claim that in 3.3V operation, a higher voltage can be gated to the V_IO pin.  One [discussion about the TMC2208 is here](https://forum.pjrc.com/threads/59827-Teensy-4-0-and-step-stipcks-hookup).  Watterout has some info on their site that is similar but not in full alignment with the PJRC posting (which implies > 3.3V on I/O lines as well). They (Watterout) sell a [protector product](https://learn.watterott.com/silentstepstick/protector/) that uses a diode to VMot to clamp over voltage.  I don't see this anywhere else. I looked through the datasheets and appnotes on the Trinamic site but didn't see anything about this.

The Watterout product also clamps the 4 stepper drive pins to VMot via diodes as well.  I assume this is to handle the case of  back-EMF. Seems odd if the Trinamic chips don't protect against that - there is no discussion of using them in the TMC2209 or TMC5160 datasheets

Can anyone shed some light on this? I could add the diode(s) to the circuit since it is cheap protection but doing so based on a random posting or 2 is a poor reason without more backup, especially if the manufacturer is silent on it.

**Comment by terjeio** [comment]
I read an article a few days back related to this, by Watterot? I did not save the link and I am not able to find it now. It went into some detail about powering Trinamics and potentional back EMF issues.

I have worked on driver code for TMC5160 recently, one thing to be aware of is that the stallGuard output is not possible to wire-or as it is with TMC2130 drivers. My Nucleo-64 board relied upon wire-or'ing beeing possible so Ihad to redesign that...

**Reply by phil-barrett** [reply]
Is [this the link](https://learn.watterott.com/silentstepstick/pinconfig/) to the article?  It talks about variable 3.3-5V stepsticks requiring power sequencing.  I don't see anything about that elsewhere but have no reason to disbelieve what they say. I am thinking that a PMOSFET might work for that.  Perhaps a high ESR cap across 3.3V makes sense as well.  

By stallGuard are you referring to the DIAG1 output on pin 17?  How did you fix it?  AND gate with pullups on inputs to allow both TMC5160 and TMC2xxx to work on the board? Or maybe a transistor to convert the signal to open collector?

**Reply by terjeio** [reply]
Yep, that is the link. I have been flipping between too many tasks lately so hard to keep track of everything.

> By stallGuard are you referring to the DIAG1 output on pin 17? How did you fix it?

Yes, I fixed it by wire'oring each with the limit switch inputs since they are driven by the opto-couplers. Not entirely ok I know - I should add a diode to protect the driver output (anode to the opto collector, cathode to driver output)? I'll have to check the MCU datasheet to see if this is ok. I have not sent the layout for a new batch so not too late for more changes.
